src/modules/secret-sharing/secret-sharing.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { AccountDAL } from '../account/account.dal';
+import { generateKey } from '../../common/utils/secretSharing';
+import { CreateSecretSharingDto } from './dtos/create-secretSharing.dto';
+import { SecretSharingDAL } from './secret-sharing.dal';
+@Injectable()
+export class SecretSharingService {
+  constructor(
+   
+    private readonly accountDAL: AccountDAL,
+   
+    private readonly secretsSharingDAL: SecretSharingDAL,
+  ) {}
+
+  async genereateKey(accountId: number) {
+    const account = await this.accountDAL.findAccountById(accountId);
```

Comment:
here you could have created a function for get account or throw error and one for key